src/main/java/org/jabref/AutomaticRelink.java

```diff
@@ -0,0 +1,58 @@
+package org.jabref;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.LinkedFile;
+
+public class AutomaticRelink {
+
+    public static void relink(BibEntry bib, String defaultFileDirectory) {
+        // Check if the file is there
+        List<LinkedFile> lfs = bib.getFiles();
+
+        // Run through all linked files
+        // Check if the files exist 
+        // Then if they don't change them
+        for (LinkedFile currentLF : lfs) {
+           Path path = Paths.get(currentLF.getLink());
+
+            if (!Files.exists(path)) {
+                // Find the File!
+                String filename = path.getFileName().toString();
+                File directory = new File(defaultFileDirectory);
+                List<String> fileLocations = new ArrayList<>();
+
+                if (directory.exists() && directory.isDirectory()) {
+                    searchFileInDirectoryAndSubdirectories(directory, filename, fileLocations);
+                    if (!fileLocations.isEmpty()) {
+                        currentLF.setLink(fileLocations.get(0));
+                    } else {
+                        System.out.println("File not found.");
```

Comment:
Please use a LOGGER. See https://devdocs.jabref.org/code-howtos/logging.html for details.

I think `debug` is appropriate here. You can even pass objects using `{}` in the string and then the object as additional paramete.r Please make use of it. Because just "File not found" doesn't help at large libraries.